homeassistant/components/modbus/base_platform.py

```diff
@@ -86,13 +88,25 @@ def __init__(self, hub: ModbusHub, entry: dict[str, Any]) -> None:
     async def async_update(self, now=None):
         """Virtual function to be overwritten."""
 
-    async def async_base_added_to_hass(self):
-        """Handle entity which will be added."""
-        if self._scan_interval > 0:
-            cancel_func = async_track_time_interval(
+    @callback
+    async def async_remote_start_stop(self, force_stop):
+        """Remote start/stop entity."""
+        if force_stop:
+            if self._cancel_timer:
+                self._cancel_timer()
+                self._cancel_timer = None
+            self._attr_available = False
+        elif self._scan_interval > 0:
+            self._cancel_timer = async_track_time_interval(
                 self.hass, self.async_update, timedelta(seconds=self._scan_interval)
             )
-            self._hub.entity_timers.append(cancel_func)
+            self._attr_available = True
+        self.async_write_ha_state()
+
+    async def async_base_added_to_hass(self):
+        """Handle entity which will be added."""
+        await self.async_remote_start_stop(False)
+        self._hub.entity_controls.append(self.async_remote_start_stop)
```

Comment:
We shouldn't store an entity interface in a shared container for outside use but use our dispatch helper to signal the entity to do the action instead.

Example:
https://github.com/home-assistant/core/blob/d2a9f7904a5e8b583d1ce41deab9919e118849b9/homeassistant/components/tuya/__init__.py#L300-L304

https://github.com/home-assistant/core/blob/d2a9f7904a5e8b583d1ce41deab9919e118849b9/homeassistant/components/tuya/__init__.py#L191

If we need to wait for the action in the dispatch caller we can use an `asyncio.Event` that the entity sets when done.

Reply:
An alternative to this whole approach is to remove the entity when disconnecting and add it again after connecting the client, and let the entity handle starting and stopping its scheduled update. That may be cleaner and keep each responsibility contained in place.

Reply:
I have now changed to use the dispatcher a MUCH cleaner solution.

However I am not sure your alternative is workable for a couple of reasons:
- It would require that we store discovery_info, in order to be able to create them again (remember this is configuration.yaml)
- Deleting the entity seems to have effects on both the UI and collection of curve data (just made a fast test), setting the entity to UNAVIALABLE seems much less intrusive. At least deleting the entity would require a lot more testing, so see there are no side effects (in scripts, data collection and UI).
- The start/stop command are to allow the user to do make the decision, there are not scheduled updates as such. It is also something that have been wished for (reload integration) just as you can do with config_flow based configurations.